### PR TITLE
Add release 1.12.5

### PR DESCRIPTION
**Pull request by JingsongLi**
Update the website to include release information of 1.12.5

### CODE REVIEW

File: _posts/2021-07-02-release-1.12.5.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+---
+layout: post
+title:  "Apache Flink 1.12.5 Released"
+date:   2021-07-02 00:00:00
+categories: news
+authors:
+- arvid:
```
Comment (leonardBang) [inline _posts/2021-07-02-release-1.12.5.md]:
```suggestion
- jingsong:
```
Should use RM name

Comment (JingsongLi) [reply _posts/2021-07-02-release-1.12.5.md]: Thanks!

**Review by leonardBang — APPROVED** [review]

File: _posts/2021-07-02-release-1.12.5.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+categories: news
+authors:
+- jingsong:
+  name: "Jingsong Lee"
+
+---
+
+The Apache Flink community released the next bugfix version of the Apache Flink 1.12 series.
+
+This release includes 21 fixes and minor improvements for Flink 1.12.4. The list below includes a detailed list of all fixes and improvements.
```
Comment (Myasuka) [inline _posts/2021-07-02-release-1.12.5.md]: I think the content should be changed from the release candidate#3

File: _config.yml [hunk 3]
```diff
@@ -79,33 +79,33 @@ flink_releases:
   -
     version_short: "1.12"
     binary_release:
-      name: "Apache Flink 1.12.4"
+      name: "Apache Flink 1.12.5"
       scala_211:
         id: "1124-download_211"
```
Comment (Myasuka) [inline _config.yml]: I think all `1124` should be changed to `1125`.

**Review by Myasuka — COMMENTED** [review]
Thanks for the update, please take a look at my comments.

File: _posts/2021-07-30-release-1.12.5.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+categories: news
+authors:
+- jingsong:
+  name: "Jingsong Lee"
+
+---
+
+The Apache Flink community released the next bugfix version of the Apache Flink 1.12 series.
+
+This release includes 80 fixes and minor improvements for Flink 1.12.4. The list below includes a detailed list of all fixes and improvements.
```
Comment (godfreyhe) [inline _posts/2021-07-30-release-1.12.5.md]: `1.12.4` -> `1.12.5`

Comment (wuchong) [reply _posts/2021-07-30-release-1.12.5.md]: `1.12.4` is correct.

File: _config.yml [hunk 5]
```diff
@@ -161,6 +161,10 @@ release_archive:
         version_short: "1.13"
         version_long: 1.13.0
         release_date: 2021-04-30
+      -
+        version_short: "1.12"
+        version_long: 1.12.5
+        release_date: 2021-07-02
```
Comment (dianfu) [inline _config.yml]: Please remember to update the release date and also the file **2021-07-30-release-1.12.5.md** during merging the PR.

**Review by dianfu — APPROVED** [review]

File: _posts/2021-07-30-release-1.12.5.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-22708'>FLINK-22708</a>] -         Propagate savepoint settings from StreamExecutionEnvironment to StreamGraph
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-22747'>FLINK-22747</a>] -         Update commons-io to 2.8
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-22757'>FLINK-22757</a>] -         Update GCS documentation
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-22774'>FLINK-22774</a>] -         Update Kinesis SQL connector&#39;s Guava to 27.0-jre
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-22799'>FLINK-22799</a>] -         Missing documentation: State backwards compatibility is not support for Flink SQL
```
Comment (carp84) [inline _posts/2021-07-30-release-1.12.5.md]: This issue has been marked as duplicated and should be removed from release note.

File: _posts/2021-07-30-release-1.12.5.md (hunk 4 above)
Comment (carp84) [reply _posts/2021-07-30-release-1.12.5.md]: The latest issue number included in this release should be 79 instead of 80 after the update.

**Review by carp84 — CHANGES_REQUESTED** [review]
Overall LGTM. Left some comments to keep up with the latest release note.

**Review by carp84 — APPROVED** [review]
+1, the latest changes LGTM. Thanks for the efforts @JingsongLi !

**Comment by JingsongLi** [comment]
Thanks all for the review~ merging...

**Pull request merged**